db/log_reader.cc

```diff
@@ -329,6 +356,54 @@ bool Reader::ReadRecord(Slice* record, std::string* scratch,
   return false;
 }
 
+void Reader::MaybeVerifyPredecessorWALInfo(
+    WALRecoveryMode wal_recovery_mode, Slice fragment,
+    const PredecessorWALInfo& expected_predecessor_wal_info) {
+  if (!track_and_verify_wals_ ||
+      wal_recovery_mode == WALRecoveryMode::kSkipAnyCorruptedRecords ||
+      stop_replay_for_corruption_) {
+    return;
+  }
+  uint64_t expected_predecessor_log_number =
+      expected_predecessor_wal_info.GetLogNumber();
+
+  // This is the first WAL recovered thus with no predecessor WAL info has been
+  // initialized
+  if (!predecessor_wal_info_.IsInitialized()) {
+    if (expected_predecessor_log_number >= min_wal_number_to_keep_) {
+      std::string reason = "Missing WAL of log number " +
+                           std::to_string(expected_predecessor_log_number);
+      ReportCorruption(fragment.size(), reason.c_str());
```

Comment:
Are all of these corruption cases covered by the unit tests?

Reply:
Let me cover all of them though the size difference is hard to cover without triggering other corruption. I will see what I can do with it.

Reply:
Fixed